cfg/rationalize_test.go

```diff
@@ -664,3 +664,47 @@ func TestRationalize_FileCacheAndBufferedReadConflict(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveLoggingConfig(t *testing.T) {
+	testCases := []struct {
+		name              string
+		config            *Config
+		expectedLogFormat string
+	}{
+		{
+			name: "valid_log_format_json",
+			config: &Config{
+				Logging: LoggingConfig{
+					Format: "json",
+				},
+			},
+			expectedLogFormat: "json",
+		},
+		{
+			name: "valid_log_format_text",
+			config: &Config{
+				Logging: LoggingConfig{
+					Format: "text",
+				},
+			},
+			expectedLogFormat: "text",
+		},
+		{
+			name: "invalid_log_format",
+			config: &Config{
+				Logging: LoggingConfig{
+					Format: "INVALID",
+				},
+			},
+			expectedLogFormat: defaultLogFormat, // Should default to JSON
```

Comment:
```suggestion
			expectedLogFormat: "json",
```

can we also handle for "TEXT" and "JSON"?

Reply:
any unintentional change to defaultLogFormat won't be caught with this test. Can we explicitly use "json" instead?